test/shared/test_server.py

```diff
@@ -0,0 +1,75 @@
+from pathlib import Path
+from typing import Mapping, Any, NamedTuple
+
+from fastcrawler.test_utils import HTTPEndpoint, Route, HTMLResponse
+from fastcrawler.engine.contracts import Response, Request
+
+
+class Homepage(HTTPEndpoint):
+    async def get(self, request, *args, **kwargs):
+        return Response(text="Hello, world!", status_code=200)
+
+
+class User(HTTPEndpoint):
+    async def get(self, request, *args, params=None, **kwargs):
+        params = params or {}
+        user_id = params.get("user_id")
+        post_id = params.get("post_id")
+        allow = params.get("allow")
+        query = params.get("query")
+        text = "Hello"
+        if user_id:
+            text += f", {user_id}"
+        if post_id:
+            text += f" in {post_id}"
+        if allow:
+            text += f", allow is {allow}"
+        if query:
+            text += f", query for {query}"
+        return Response(text=text, status_code=200)
```

Comment:
While this is good and very flexible, but i'd also want to have a builder pattern here
ResponseBuilder(status=200, methods="GET")
this way, while user have the ability to be flexible, he can also be fast at mocking APIs.